### PR TITLE
Update README with links

### PR DESCRIPTION
**Pull request by gustawlippa**
This PR adds links to relevant files, and removes mention of `doc/README` which does not exist.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ See the documentation for the latest releases:
 
 **MongooseIM documentation highlights:**
 
-When developing new features/modules, please make sure you add basic documentation to the 'doc/' directory, and add a link to your document in 'doc/README.md.'
+When developing new features/modules, please make sure you add basic documentation to the ['doc/'](https://github.com/esl/MongooseIM/tree/master/doc) directory, and add a link to your document in ['mkdocs.yml'](https://github.com/esl/MongooseIM/blob/master/mkdocs.yml).
```
Comment (vkatsuba) [inline README.md:79]:
```suggestion
When developing new features/modules, please make sure you add basic documentation to the ['doc'](doc) directory, and add a link to your document in ['mkdocs.yml'](mkdocs.yml).
```

Comment (gustawlippa) [reply README.md:79]: Ah, true, it's from GH to GH - too much looking at the docs.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/76274) / small_tests / 97fcfbddeb38761a3ee9e2950582ca610dd02083
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3306/76274/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3306/76274/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-28_11.52.09/lib.mongooseim.logs/run.2021-09-28_11.52.10/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/76295) / small_tests / 4fc804a90c0059ddb42f453d60b1d0092550c467
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3306/76295/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3306/76295/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-28_12.02.32/lib.mongooseim.logs/run.2021-09-28_12.02.32/suite.log.html)

**Review by vkatsuba — APPROVED** [review]

**Pull request merged**